src/anndata/_core/sparse_dataset.py

```diff
@@ -660,13 +663,19 @@ class _CSCDataset(BaseCompressedSparseDataset, abc.CSCDataset):
     """Internal concrete version of :class:`anndata.abc.CSRDataset`."""
 
 
-def sparse_dataset(group: GroupStorageType) -> abc.CSRDataset | abc.CSCDataset:
+def sparse_dataset(
+    group: GroupStorageType,
+    *,
+    should_cache_indptr: bool = True,
```

Comment:
This is the only place this appears as a "public" api.  We could just use the private variable directly by assignment.